lib/elastic_apm/metadata/system_info.rb

```diff
@@ -50,7 +50,7 @@ def gem_platform
       private
 
       def detect_hostname
-        `hostname`.chomp
+        Socket.gethostname.chomp
```

Comment:
Do we know whether it's already chomped by Socket?

Reply:
And does it ever raise?

Reply:
I believe it can, looking at the c source [here](https://ruby-doc.org/stdlib-2.5.3/libdoc/socket/rdoc/Socket.html#method-c-gethostname)
I'll add in error handling

Reply:
@mikker I updated this. Looking at the c source, `chomp` is not called on the hostname and all the possible errors raised are `StandardErrors` so I just `rescue` and return the empty string.